### PR TITLE
M2P-151 Add feature switch to handle with virtual products as physical

### PR DESCRIPTION
**Pull request by vitaliyreznikov**
In current Bolt checkout flow, the shipping and tax endpoint is not called for virtual carts,
It means we don't support taxes for virtual product.
In this PR we add feature switch to handle all virtual products as physical, so it's a workaround for known issue.
 and should handle all products as physical

Fixes: [M2P-151]

#changelog M2P-151 Add feature switch to handle with virtual products as physical

# Type of change

- [x] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# For PR Reviewer 
- [ ] Reviewed unit tests to make sure we are using real components rather than mocks as much as possible?
- [ ] For any major change (observer, new Bolt feature, core Magento interaction) we must add a feature switch, did you verify this?

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have created or modified unit tests to sufficiently cover my changes.
- [x] I have added my Jira ticket link and provided a changelog message.

### CODE REVIEW

File: Helper/Cart.php [hunk 1]
```diff
@@ -1363,6 +1363,15 @@ function ($item) use ($imageHelper, &$totalAmount, &$diff, $storeId, $currencyCo
             $products[] = $product;
         }
 
+        if ($this->deciderHelper->isHandleVirtualProductsAsPhysical()) {
```
Comment (danac-gs) [inline Helper/Cart.php:1366]: You could have added this condition in the initial loop, to the line 1278 above.

**Comment by vitaliyreznikov** [comment]
Thank you @danac-gs . Updated. Could you please review again?

File: Helper/FeatureSwitch/Decider.php [hunk 2]
```diff
@@ -257,4 +257,8 @@ public function isSaveCartInSections() {
     public function ifShouldDisablePrefillAddressForLoggedInCustomer() {
         return $this->isSwitchEnabled(Definitions::M2_IF_SHOULD_DISABLE_PREFILL_ADDRESS_FROM_BOLT_FOR_LOGGED_IN_CUSTOMER);
     }
+
+    public function isHandleVirtualProductsAsPhysical() {
```
Comment (matt-thomason) [inline Helper/FeatureSwitch/Decider.php:261]: I think the function signature should match the constant. Either `handleVirtualProductsAsPhysical` or change the constant to `M2_IS_VIRTUAL_PRODUCTS_HANDLED_AS_PHYSICAL` (reworded to sound a bit better).

I think I prefer the former but it's up to you.

Comment (vitaliyreznikov) [reply Helper/FeatureSwitch/Decider.php]: Done

**Pull request merged**